src/garage/tf/algos/npo_task_embedding.py

```diff
@@ -0,0 +1,1349 @@
+from dowel import Histogram, logger, tabular
+import numpy as np
+import tensorflow as tf
+import scipy.stats
+
+from garage import log_performance, TaskEmbeddingTrajectoryBatch
+from garage.misc import tensor_utils as np_tensor_utils
+from garage.tf.algos.batch_polopt import BatchPolopt
+from garage.tf.embeddings import StochasticEmbedding
+from garage.tf.misc.tensor_utils import center_advs
+from garage.tf.misc.tensor_utils import compile_function
+from garage.tf.misc.tensor_utils import compute_advantages
+from garage.tf.misc.tensor_utils import concat_tensor_list
+from garage.tf.misc.tensor_utils import discounted_returns
+from garage.tf.misc.tensor_utils import filter_valids
+from garage.tf.misc.tensor_utils import filter_valids_dict
+from garage.tf.misc.tensor_utils import flatten_batch
+from garage.tf.misc.tensor_utils import flatten_batch_dict
+from garage.tf.misc.tensor_utils import flatten_inputs
+from garage.tf.misc.tensor_utils import graph_inputs
+from garage.tf.misc.tensor_utils import new_tensor
+from garage.tf.misc.tensor_utils import pad_tensor
+from garage.tf.misc.tensor_utils import pad_tensor_n
+from garage.tf.misc.tensor_utils import pad_tensor_dict
+from garage.tf.misc.tensor_utils import positive_advs
+from garage.tf.misc.tensor_utils import stack_tensor_dict_list
+from garage.tf.misc.tensor_utils import stack_tensor_list
+from garage.tf.optimizers import LbfgsOptimizer
+from garage.tf.policies import StochasticMultitaskPolicy
+
+
+class NPOTaskEmbedding(BatchPolopt):
```

Comment:
Having a complete copy of NPO is a serious issue. These duplicates never stay in sync, and result in a big maintenance burden.

Please see RL2's changes for alternate ways of implementing this.